src/raftstore/store/worker/pd.rs

```diff
@@ -174,6 +182,57 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_validate_peer(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
+            Ok(mut pd_region) => {
+                metric_incr!("pd.get_region");
+                if pd_region == local_region {
+                    // Local region info is as fresh as pd, which means
+                    // local peer is still in the region, let it be there.
+                    info!("peer {:?} is still valid because local region info is as fresh as pd",
+                          peer);
+                    return;
+                }
+                if !(pd_region.get_start_key() == local_region.get_start_key()) {
```

Comment:
If we support `merge` later, the check is not correct here.
Checking whether pd_region's range covers the search key or not may be better.